### DISCUSSION TITLE
Open html files in browser in 3.4.6

### DISCUSSION THREAD

**Discussion by mplaut** [sumatrapdfreader/sumatrapdf#2932] [Documentation]
I recently upgraded to 3.4.6. In my old 3.3.x version, when I clicked on a link to an HTM file inside a PDF it automatically opened it in a browser. Now in 3.4.6 it opens the HTM file in Sumatra. 

How can I reproduce the old behavior?

**Comment by GitHubRulesOK** [comment]
wow 7.x got a time machine ? currently only 3.5pr is available perhaps roll back to 3.3 ? but without sample there is no way to know which version of SumatraPDF has what ability

**Comment by mplaut** [comment]
Sorry. I guess I was distracted. I meant 3.4.6

**Comment by mplaut** [comment]
OK. I have done more homework. 
Version 3.1.2 seems to be the last version that behaves the way I need it to behave out of the box. When I click on an HTM file it just automatically opens it in the default browser.
Version 3.2 initiates a new behavior that seems to persist through all versions of 3.3, namely, when I click on an HTM file it says: Error loading (the full pathname). 
In version 3.4.6 clicking on the same link in the same file, that is a link to an HTML file, just opens it in Sumatra. Since it is an HTML file it obviously does not look so good. 

I would like to be able to revert to the behavior of 3.1.2  Is there a setting that I can turn on or off? I read through the Settings documentation and nothing seemed likely. 

I also thought to at least set up a browser as an ExternalViewer for HTML files but that also does not seem to work for me. 

ExternalViewers [
	[
		CommandLine = "c:\Program Files\Mozilla Firefox\firefox.exe" "%1"
		Name = FireFox
		Filter = *.htm;*.html
	]
]

It seems that it will only accept ExternalViewers for one of the file types that Sumatra opens itself, but not for HTML. So I tried adding *.PDF to the Filter of firefox and then it appeared in the File menu and when I invoked the menu item for an HTML file read into Sumatra it worked. But this is obviously not ideal. 

Any help much appreciated.

**Reply by GitHubRulesOK** [reply]
without your sample i cant help other than

![image](https://user-images.githubusercontent.com/12754913/188521846-5a6880c5-fc7b-4f61-9daf-d97b38727f0a.png)

```
ExternalViewers [
	[
		CommandLine = "C:\\apps\Browsers\Firefox 97.0\core97x32\firefox.exe"  "file:///%1#page=%p"
		Name = FireFox
		Filter = *.htm;*.html;*.pdf
	]
]
```
I use file/// as usually faster but may not work in all cases, you may not want #page=%p as it only applies to paged pdf (html is one page)
I agree html looks nothing like "as good as" in a browser and even the section number is missing
There are ways to search a phrase in firefox that potentially SumatraPDF should find possible however the best for this use is copy a chunk of key word(s) to clip and use firefox F3 to find that block

![image](https://user-images.githubusercontent.com/12754913/188521062-29131c31-94ed-4687-8897-6cc388a5840a.png)


HTH

**Comment by mplaut** [comment]
My HTML files are usually in a different language and are not really visible at all in Sumatra. I have no need to jump to a page. Just to open the file at the beginning. 

I am attaching a sample to demonstrate the problem. 
1. Make a directory Archive in the root.
2. Unzip the attached file. 
3. Load the file TextsTest.pdf  This is the master that loads the other 2. 
4. TestPDF loads as it shou;d.
5. TestHTML loads into Sumatra and can then be opened in Firefox from the files menu. It used to be loaded directly into the default browser in 3.1

Can I make a keyboard shortcut/hotkey to open the ExternalViewer?
Is there something I can set to make Sumatra open it directly into the browser?
Can I set the default browser as the ExternalViewer rather than Firefox specifically?

By the way, I think Sumatra is great and have been using it for many years and telling many others.

Thanks in Advance.

**Reply by GitHubRulesOK** [reply]
attachment missing
currently don't think there is a way to set single key only Alt + F + your choice of key in Name = Firefo&x the hot key is `Alt+F+X` since `Alt+F+F` is = English `show current file in folder` (perhaps useful for fast navigate)

File type defaults are a system choice so my default double click or run file.htm defaults to edge but could be altered via properties 
![image](https://user-images.githubusercontent.com/12754913/188651104-32230da9-7439-4cb3-b38d-32ecbd1a01b6.png)

**Comment by mplaut** [comment]
Sorry. Here is the attachment.
[TestLinks.zip](https://github.com/sumatrapdfreader/sumatrapdf/files/9498013/TestLinks.zip)

**Reply by GitHubRulesOK** [reply]
ok that was a bit confusing by relative links but your core issue is

 open TextsTest.pdf and click top right link opens File: TestHTML.html

agreed its new pdf/html engine behavior and the html rendering is very poor

currently see no workaround other than mentioned so far (agreed a choice of open link externally would be useful)
@kjk comments ?

**Comment by mplaut** [comment]
Do you mean by your reply that Sumatra now has a builtin HTML rendering engine, and that is why it loads the HTML file into itself, while before it called the default external app? Will it in the future support non-English? On what basis will the HTML rendering engine be improved?

Alt+f+x is not such a good choice since "x" is also for Exit. If I use Alt+f+i it executes immediately.

**Reply by GitHubRulesOK** [reply]
3.2+ uses more MuPDF epub/htmlx features than before so the effect is that html xml fb2 etc rendering support is very much increasingly dictated by MuPDF updates.

File handling is more down to how SumatraPDF passes those objects, thus my user observations can only be changed by the developer kjk

**Comment by mplaut** [comment]
Your comment about MuPDF was interesting and informative. I see that the support of MuPDF for HTML is still pretty primitive. Although regular English HTML files look much better than my Hebrew files, they still do not support HTML tables though this is planned. 

In view of this very incomplete HTML support in MuPDF it does seem to me a good idea to be able to avoid it. 

Also I like the searching in browsers that shows how many instances of the search string there are in the whole file. It works very fast even for large files. 

Thank you for your comments.
